src/hotspot/cpu/x86/sharedRuntime_x86.cpp

```diff
@@ -62,9 +62,11 @@ void SharedRuntime::inline_check_hashcode_from_object_header(MacroAssembler* mas
 
 
   if (LockingMode == LM_LIGHTWEIGHT) {
-    // check if monitor
-    __ testptr(result, markWord::monitor_value);
-    __ jcc(Assembler::notZero, slowCase);
+    if (!UseObjectMonitorTable) {
+      // check if monitor
+      __ testptr(result, markWord::monitor_value);
+      __ jcc(Assembler::notZero, slowCase);
+    }
```

Comment:
Did I miss this one?

Reply:
We did the C2 one. This is the native wrapper one. We both overlooked it.

Reply:
I see.  We missed an optimization here.